pkg/monitor/cluster/podconditions.go

```diff
@@ -120,3 +123,33 @@ func (mon *Monitor) _emitPodContainerStatuses(ps *corev1.PodList) {
 		}
 	}
 }
+
+func (mon *Monitor) _emitPodContainerRestartCounter(ps *corev1.PodList) {
+	for _, p := range ps.Items {
+		if !namespace.IsOpenShiftSystemNamespace(p.Namespace) {
+			continue
+		}
+
+		for _, cs := range p.Status.ContainerStatuses {
+			if cs.RestartCount < restartCounterThreadshold {
+				continue
+			}
+
+			mon.emitGauge("pod.restartcounter", int64(cs.RestartCount), map[string]string{
+				"name":          p.Name,
+				"namespace":     p.Namespace,
+				"nodeName":      p.Spec.NodeName,
+				"containername": cs.Name,
```

Comment:
For consistency with other fields.

```suggestion
				"containerName": cs.Name,
```

Same with below in the logger.